### ISSUE TITLE
Disallow TestInstanceFactory overrides in @Nested test classes

### ISSUE THREAD

**Issue by sbrannen** [junit-team/junit5#1573]
## Overview

#672 introduced a new `TestInstanceFactory` extension API with support for overriding a `TestInstanceFactory` registered on an enclosing class for an enclosed `@Nested` class. However, similar "overriding" support does not exist for traditional test class hierarchies (i.e., a subclass cannot override a `TestInstanceFactory` registered on a superclass).

_Team Decision_ from https://github.com/junit-team/junit5/issues/1567#issuecomment-417624234:

> Align the behavior between conventional class hierarchies and nested class structures with regard to disallowing multiple registrations of `TestInstanceFactory` extensions. Consequently, it should not be possible to override a `TestInstanceFactory` registered on an enclosing class for an enclosed `@Nested` test class.

## Deliverables

- [x] Disallow `TestInstanceFactory` overrides in `@Nested` test classes.
- [x] Update JavaDoc accordingly.
- [x] Update the User Guide accordingly.

**Comment by sbrannen** [comment]
_in progress_

**Comment by seanf** [comment]
@sbrannen So if I'm understanding correctly, this will make the old version of the test `org.junit.jupiter.engine.extension.TestInstanceFactoryTests#instanceFactoriesInNestedClassHierarchy` (`org.junit.jupiter.engine.extension.TestInstanceFactoryTests.OuterTestCase`) illegal?

```java
	@ExtendWith(FooInstanceFactory.class)
	static class OuterTestCase {
		@Nested
		@ExtendWith(NestedInstanceFactory.class)
		class InnerTestCase {
```
```

**Comment by sbrannen** [comment]
This has been addressed in `master` in 16ac9ac4ded360dd33f2e55cfa3adbfeee33ca4b.

**Comment by sbrannen** [comment]
@seanf, yes that's correct. See the updated tests in the aforementioned commit.
